### PR TITLE
feat: add WOOFiDexCrossChainRouter

### PR DESCRIPTION
**Pull request by merlincdj**

### CODE REVIEW

File: contracts/interfaces/WOOFiDex/INonceCounter.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.14;
+
+interface INonceCounter {
+    /* ----- Events ----- */
+
+    event CrossChainRouterUpdated(address crossChainRouter, bool flag);
+
+    /* ----- Functions ----- */
+
```
Comment (fb-alexcq) [inline contracts/interfaces/WOOFiDex/INonceCounter.sol:10]: As an interface, for the API's completeness, maybe add 
`inboundNonce` function ?

File: contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+            toToken,
+            minToAmount,
+            woofiDexVault,
+            dstVaultDeposit
+        );
+    }
+
+    function quoteLayerZeroFee(
+        address to,
+        SrcInfos calldata, // srcInfos
```
Comment (fb-alexcq) [inline contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol:191]: Better to remove this line ?

File: contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+            address,
+            uint256,
+            DstVaultDeposit memory
+        )
+    {
+        (uint256 nonce, address to, address toToken, uint256 minToAmount, DstVaultDeposit memory dstVaultDeposit) = abi
+            .decode(payload, (uint256, address, address, uint256, DstVaultDeposit));
+
+        return (nonce, to, toToken, minToAmount, dstVaultDeposit);
+    }
```
Comment (fb-alexcq) [inline contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol:306]: No need for these two one-line methods

File: contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+                    bridgedAmount,
+                    toToken,
+                    minToAmount,
+                    toToken,
+                    realToAmount,
+                    vaultDepositFE.accountId,
+                    vaultDepositFE.brokerHash,
+                    vaultDepositFE.tokenHash,
+                    vaultDepositFE.tokenAmount
+                );
```
Comment (fb-alexcq) [inline contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol:416]: Code redundancy with the above logic?

**Review by fb-alexcq — COMMENTED** [review]
Almost there.

File: contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+                bridgedToken,
+                bridgedAmount,
+                toToken,
+                minToAmount,
+                toToken,
+                bridgedAmount,
+                vaultDepositFE.accountId,
+                vaultDepositFE.brokerHash,
+                vaultDepositFE.tokenHash,
+                vaultDepositFE.tokenAmount
```
Comment (fb-alexcq) [inline contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol:386]: Use `vaultDepositFE` struct as a whole?

File: contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol (hunk 2 above)
Comment (merlincdj) [reply contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol:191]: agree

File: contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol [hunk 6]
```diff
@@ -348,6 +316,25 @@ contract WOOFiDexCrossChainRouter is IWOOFiDexCrossChainRouter, Ownable, Pausabl
         );
     }
 
+    function _depositTo(
+        address to,
+        address toToken,
+        address woofiDexVault,
+        DstVaultDeposit memory dstVaultDeposit,
+        uint256 tokenAmount
+    ) internal returns (IWOOFiDexVault.VaultDepositFE memory) {
```
Comment (fb-alexcq) [inline contracts/WOOFiDex/WOOFiDexCrossChainRouter.sol:325]: No need for the return value.

**Review by fb-alexcq — APPROVED** [review]

**Pull request merged**